pkg/plugin/GlobalPluginService.go

```diff
@@ -1948,3 +1930,330 @@ func (impl *GlobalPluginServiceImpl) MigratePluginDataToParentPluginMetadata(plu
 	}
 	return nil
 }
+
+func (impl *GlobalPluginServiceImpl) GetAllPluginMinData() ([]*bean2.PluginMinDto, error) {
+	pluginsParentMinData, err := impl.globalPluginRepository.GetAllPluginMinData()
+	if err != nil {
+		impl.logger.Errorw("GetAllPluginMinData, error in getting all plugin parent metadata min data", "err", err)
+		return nil, err
+	}
+	pluginMinList := make([]*bean2.PluginMinDto, 0, len(pluginsParentMinData))
+	for _, item := range pluginsParentMinData {
+		if item.Type == repository.PLUGIN_TYPE_PRESET {
+			continue
+		}
+		pluginMinList = append(pluginMinList, bean2.NewPluginMinDto().WithParentPluginId(item.Id).WithPluginName(item.Name).WithIcon(item.Icon))
+	}
+	return pluginMinList, nil
+}
+
+func (impl *GlobalPluginServiceImpl) checkValidationOnPluginNameAndIdentifier(pluginReq *bean2.PluginParentMetadataDto) error {
+	plugins, err := impl.globalPluginRepository.GetAllPluginMinData()
+	if err != nil {
+		impl.logger.Errorw("error in getting all plugins", "err", err)
+		return err
+	}
+	for _, plugin := range plugins {
+		if plugin.Identifier == pluginReq.PluginIdentifier {
+			return &util.ApiError{
+				HttpStatusCode:  http.StatusConflict,
+				Code:            strconv.Itoa(http.StatusConflict),
+				InternalMessage: bean2.PluginWithSameIdentifierExistsError,
+				UserMessage:     bean2.PluginWithSameIdentifierExistsError,
+			}
+		}
+		if plugin.Name == pluginReq.Name {
+			return &util.ApiError{
+				HttpStatusCode:  http.StatusConflict,
+				Code:            strconv.Itoa(http.StatusConflict),
+				InternalMessage: bean2.PluginWithSameNameExistError,
+				UserMessage:     bean2.PluginWithSameNameExistError,
+			}
+		}
+	}
+	return nil
+}
+
+func (impl *GlobalPluginServiceImpl) checkValidationOnVersion(pluginReq *bean2.PluginParentMetadataDto) error {
+	pluginVersions, err := impl.globalPluginRepository.GetPluginVersionsByParentId(pluginReq.Id)
+	if err != nil {
+		impl.logger.Errorw("checkValidationOnVersion, error in getting all plugins versions by parentPluginId", "parentPluginId", pluginReq.Id, "err", err)
+		return err
+	}
+	for _, pluginVersion := range pluginVersions {
+		if pluginVersion.PluginVersion == pluginReq.Versions.DetailedPluginVersionData[0].Version {
```

Comment:
add size check for panic handling

Reply:
add comment on the if condition

Reply:
actually these are already handled in validations v2 func, which sits on top when service layer func in called